src/DropdownReference/components/DropdownReferenceContainer.ts

```diff
@@ -0,0 +1,252 @@
+        .then((newValue: any) => {
+            if (newValue) {
+                this.setState({
+                    isClearable: this.props.isClearable ? true : false,
+                    selectedObject: {
+                        value: newValue.getGuid(),
+                        label: mx.parser.formatAttribute(newValue, props.attribute)
+                    }
+                });
+            } else {
+                this.setState({
+                    isClearable: false,
+                    selectedObject: {
+                        value: "",
+                        label: ""
+                    }
+                });
+            }
+        });
+    }
+
+    private isReadOnly = (): boolean => {
+        const { editable, mxObject, readOnly } = this.props;
+
+        return editable !== "default" || !mxObject || readOnly;
+    }
+
+    private resetSubscriptions(mxObject?: mendix.lib.MxObject) {
+        this.subscriptionHandles.forEach(window.mx.data.unsubscribe);
+        this.subscriptionHandles = [];
+
+        if (mxObject) {
+            this.subscriptionHandles.push(window.mx.data.subscribe({
+                callback: this.handleSubscriptions,
+                guid: mxObject.getGuid()
+            }));
+            this.subscriptionHandles.push(window.mx.data.subscribe({
+                attr: this.association,
+                callback: this.handleSubscriptions,
+                guid: mxObject.getGuid()
+            }));
+            this.subscriptionHandles.push(window.mx.data.subscribe({
+                attr: this.props.attribute,
+                callback: this.handleSubscriptions,
+                guid: mxObject.get(this.association) as string
+            }));
+            this.subscriptionHandles.push(window.mx.data.subscribe({
+                entity: this.props.entityPath.split("/")[1],
+                callback: () => {
+                    this.setState({ selectedObject: null });
+                    this.getSelectedValue(this.props);
+                    this.retrieveOptions(this.props);
+                }
+            }));
+        }
+    }
+
+    private handleSubscriptions = () => {
+        this.getSelectedValue(this.props);
+    }
+
+    private onChange = (selection: ReferenceOption) => {
+        if (this.props.mxObject) {
+
+            if (!selection) {
+                const references = this.props.mxObject.getReferences(this.association);
+                this.props.mxObject.removeReferences(this.association, references);
+            } else {
+                this.props.mxObject.set(this.association, selection.value);
+                if (!this.state.selectedObject || (this.state.selectedObject.value !== selection.value)) {
+                    this.executeOnChangeEvent();
+                }
+            }
+        }
+
+        this.setState({ selectedObject: selection ? selection : null });
+    }
+
+    private executeOnChangeEvent = () => {
```

Comment:
No need for function property 
```suggestion
    private executeOnChangeEvent() {
```